Automattic/wp-calypso
# Earn: Refactor plans modal to have own css

## Proposed Changes

- This is a requested follow up to another PR. See comment [here](https://github.com/Automattic/wp-calypso/pull/80296#discussion_r1293867603).
- This PR refactors the RecurringPaymentsPlanAddEditModal to have it's own CSS. Before, CSS for the component was included with styles for memberships page. But that means if we import this component to use elsewhere, as we are now doing with Launchpad, the CSS doesn't come with the component. We were needing to duplicate some styles. After this PR, css for the component is tied to and travels with the component. 
- In line with calypso convention, since this is its own re-usable component with it's own css, I moved the component file (and the new css file) to it's own folder. This accounts for the change looking bigger than it actually is or was probably expected to be.

## Testing Instructions



Behavior before and after this PR should be exactly the same. We're just testing to ensure there are no regressive breakages.

1) You will need a test site that has a Stripe connection.
2) Test the Earn page
   - Check out this branch and go to http://calypso.localhost:3000/earn/payments-plans/YOURDOMAIN
   - Click the `Add a new payment plan` button, add a plan, and confirm the plan is added after the modal closes. 
   - Click to edit the modal, and try changing various options and just be sure they all save as expected. 
3) Test Launchpad
   - Start a newsletter site at http://calypso.localhost:3000/setup/newsletter/intro, choose the Paid option, and proceed to Launchpad. You can also use another newsletter site, with paid newsletter, that still has full screen launchpad if you have one.
   - If needed, connect Stripe. If you want to avoid fully connecting Stripe, add `define( 'USE_STORE_SANDBOX', true ); `to wp-content/mu-plugins/0-sandbox.php in your sandbox, and sandbox public-api.wordpress.com and the domain of your test site. Then click the link to go to Stripe and you'll see a button to skip the setup. When done, you should return to Launchpad. 
   - Click the Launchpad task to `Create paid newsletter`. Confirm the Plans Modal opens and looks like it is supposed to look. Create a plan, save, and confirm you see a success notification in the upper right corner of your screen. Navigate to the Earn page and confirm the plan you just created is visible on your plans page.

## Pre-merge Checklist
